src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -227,6 +230,7 @@ private EventsRetryResult retryEventsOnce(String userId, String deviceId, List<E
         break;
       case PAYLOAD_TOO_LARGE:
         shouldRetry = true;
+        shouldReduceEventCount = true;
```

Comment:
Do we need to reduceEventCount if the payload is too large?

Reply:
I saw in node sdk. It's set to true.https://github.com/amplitude/Amplitude-Node/blob/main/packages/node/src/retry/defaultRetry.ts
And since response indicate payload is too large. Retry without reduct events make no sense.